test/test_encryption.py

```diff
@@ -605,14 +606,19 @@ def parse_auto_encrypt_opts(self, opts):
             opts["kms_tls_options"] = KMS_TLS_OPTS
         if "key_vault_namespace" not in opts:
             opts["key_vault_namespace"] = "keyvault.datakeys"
+        if encrypted_fields:
+            opts["encrypted_fields_map"] = encrypted_fields
+
         opts = dict(opts)
         return AutoEncryptionOpts(**opts)
 
-    def parse_client_options(self, opts):
+    def parse_client_options(self, opts, encrypted_fields=None):
         """Override clientOptions parsing to support autoEncryptOpts."""
         encrypt_opts = opts.pop("autoEncryptOpts")
-        if encrypt_opts:
-            opts["auto_encryption_opts"] = self.parse_auto_encrypt_opts(encrypt_opts)
+        if encrypt_opts or encrypted_fields:
+            opts["auto_encryption_opts"] = self.parse_auto_encrypt_opts(
+                encrypt_opts, encrypted_fields=encrypted_fields
+            )
```

Comment:
From my reading of the spec test, a top-level `encrypted_fields` only means that the test setup code should pass encrypted_fields to create_collection. I don't think it means we pass it to auto_encryption_opts. I think the expected behavior should be similar to how the json_shema field is handled in setup_scenario().